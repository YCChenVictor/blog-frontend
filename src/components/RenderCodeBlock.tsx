import React from 'react';
import {Prism as SyntaxHighlighter} from 'react-syntax-highlighter'

const RenderCodeBlock = (props: {
  inline?: boolean;
  className?: string;
  children: React.ReactNode;
}) => {
  let language;
  let result;
  if (props.inline === true || !props.className) {
    language = 'inline';
  } else {
    const match = /language-(\w+)/.exec(props.className || '');
    language = match ? match[1] : 'inline';
  }

  if (language === 'inline') {
    result = (
      <code className="bg-gray-500 text-white p-0.5">{props.children}</code>
    );
  } else {
    result =
    <SyntaxHighlighter language={language} wrapLines={true} lineProps={{style: {wordBreak: 'break-all', whiteSpace: 'pre-wrap'}}} PreTag="div">
      {props.children as string[]}
    </SyntaxHighlighter>
  }

  return result;
};

export default RenderCodeBlock;
