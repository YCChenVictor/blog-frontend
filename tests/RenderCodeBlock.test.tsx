import * as renderer from "react-test-renderer";
import * as React from "react";
import RenderCodeBlock from "../src/components/RenderCodeBlock";

describe("App", () => {
  it("renders RenderCodeBlock without crashing", () => {
    const component = renderer.create(<RenderCodeBlock children={""} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
