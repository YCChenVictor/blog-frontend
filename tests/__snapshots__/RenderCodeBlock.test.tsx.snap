// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders RenderCodeBlock without crashing 1`] = `
<code
  className="bg-gray-500 text-white p-0.5"
/>
`;
